### ISSUE TITLE
Calling Init() on ViewModel creates exception without parameters

### ISSUE THREAD

**Issue by indazoo** [MvvmCross/MvvmCross#1132]
Hi

I found a small bug(?) in Cirrious.MvvmCross.ViewModels.CallBundleMethod().
If the parameter "bundle" is null, then CallBundleMethod(...) fails. This creates problems for me because this case occurs when I try to use design time data in Visual Studio. The error does not occur at runtime.
I'm not very familiar with the inner MVVMcross source code and do not really know at what place the check should be done. But with an additional line to check for bundle == null, the Designer no longer throws an exception.
Maybe the check should be much earlier in the stack because if there is no bundle there is no need to run CallBundleMethod and it should fail (don't no your coding style). Maybe there should be a IsDesigner check earlier? I found no way to intercept/override a method to do a designer check in my code after Start().

Anyway, that helped:
public static void CallBundleMethod(this IMvxViewModel viewModel, MethodInfo methodInfo, IMvxBundle bundle)
        {
            if (bundle == null) return;  //!!!! new line , DesignMode in Visual Studio needs this check
                        var parameters = methodInfo.GetParameters().ToArray(); 

I created a small visual studio project with latest MvvmCross code (a few days old):
https://github.com/indazoo/MvvmCross_DesignData/commits/master
There are two commits of today. The earlier one throws an Exception in Designer.

Your help would be greatly appreciated.

P.S. There is some confusion on my side about windows.Common and Windows Store and when to use what. The samples on the net and your fast progress with the renaming/moving (lately the plugins) produce a steep learning curve on my side. Maybe I miss something.

**Comment by indazoo** [comment]
This MvvmCross thing is really good! ;-)  I start liking it!
I overrode MvxDefaultViewModelLocator in App.cs as seen here:
https://github.com/MvvmCross/MvvmCross/wiki/Customising-using-App-and-Setup
This works, but has the drawback is that I have to check for every new MvvmCross version if the code is valid/the same. If the check can be implemented into MvvmCross It would be great.
```
	using Cirrious.MvvmCross.ViewModels;
	using Cirrious.CrossCore;
	using Cirrious.CrossCore.Exceptions;

	public class DesignTimeViewModelLocator : MvxDefaultViewModelLocator
	{
		public override IMvxViewModel Load(Type viewModelType,
									IMvxBundle parameterValues,
									IMvxBundle savedState)
		{
			IMvxViewModel viewModel;
			try
			{
				viewModel = (IMvxViewModel)Mvx.IocConstruct(viewModelType);
			}
			catch (Exception exception)
			{
				throw new MvxException(exception, "Problem creating viewModel of type {0}", viewModelType.Name);
			}

			try
			{
				if (!App2.Helpers.DesignTime.IsInDesignMode) //My design time checker.
				{
					CallCustomInitMethods(viewModel, parameterValues);
					if (savedState != null)
					{
						CallReloadStateMethods(viewModel, savedState);
					}
				}
				viewModel.Start();
			}
			catch (Exception exception)
			{
				throw new MvxException(exception, "Problem initialising viewModel of type {0}", viewModelType.Name);
			}

			return viewModel;
		}
	}
```

**Comment by martijn00** [comment]
If you want you could create a pull request to add your code to the source of MvvmCross. Then we will review it to see if it is correct.

**Comment by indazoo** [comment]
Thanks for your response,
I would, but I do not have a clue how you want to have the detecting of the different Designers implemented in your code. For me it was easy. I have a small static method (App2.Helpers.DesignTime.IsInDesignMode) in my code. Your code is highly modularized and I have no idea if it is mandatory to use some classes like MvxDesignTimeHelper, or to be honest how to implement correctly with your IOC stuff and then the mother of all questions : Where? This is not criticism, you guys are just miles ahead of me and I just started using MvvmCross. You know exactly where this stuff belongs.
And, after my contributions in other projects being rejected by things like "we don't like your formatting of your comments" I'm a bit "contribute shy" ! So I posted the code to help. If you think no one else uses that just close the issue, its ok for me. If you think it is needed put it on low fire for the rainy/cold winter.
